### PR TITLE
survey block statuses tab updates

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
*Issue #:*
https://github.com/bcgov/met-public/issues/397

*Description of changes:*
Changes made to 
- insert status blocks data to engagement status block table in case the default values are updated during engagement creation
- update status blocks data for an engagement for any changes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/engagement/form/AddSurveyBlockTabs.tsx [hunk 1]
```diff
@@ -1,14 +1,47 @@
...
+        });
+        setcontentList(updatedSurveyBlockContent);
+    };
+
+    // exclude content having empty block text
+    useEffect(() => {
+        const filterContentList = contentList.filter(function (el) {
+            return el.block_text != '';
+        });
+        handleChange(filterContentList);
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/form/AddSurveyBlockTabs.tsx:42]: Do we need to manage the state in two places, here and in the context file?

File: met-web/src/components/engagement/form/AddSurveyBlockTabs.tsx [hunk 2]
```diff
@@ -25,13 +58,19 @@ const AddSurveyBlockTabs = () => {
...
                 <MetTabPanel value={SubmissionStatus[SubmissionStatus.Upcoming]}>
-                    <RichTextEditor initialHTMLText={upcomingText} />
+                    <RichTextEditor
+                        handleEditorStateChange={handleRichDescriptionChange}
+                        initialHTMLText={upcomingText}
+                    />
                 </MetTabPanel>
                 <MetTabPanel value={SubmissionStatus[SubmissionStatus.Open]}>
-                    <RichTextEditor initialHTMLText={openText} />
+                    <RichTextEditor handleEditorStateChange={handleRichDescriptionChange} initialHTMLText={openText} />
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/form/AddSurveyBlockTabs.tsx:67]: handleRichDescriptionChange? The description field is another field in engagement form, this doesn't change that right?

File: met-web/src/components/engagement/form/AddSurveyBlockTabs.tsx [hunk 3]
```diff
@@ -1,14 +1,47 @@
...
+    // set initial value to upcoming
     const [value, setValue] = React.useState(SubmissionStatus[SubmissionStatus.Upcoming]);
 
+    // array to pass updated content to engagement context
+    const surveyBlockContent = [
+        { survey_status: SubmissionStatus[SubmissionStatus.Upcoming], block_text: '' },
+        { survey_status: SubmissionStatus[SubmissionStatus.Open], block_text: '' },
+        { survey_status: SubmissionStatus[SubmissionStatus.Closed], block_text: '' },
+    ];
+    const [contentList, setcontentList] = useState(surveyBlockContent);
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/form/AddSurveyBlockTabs.tsx:24]: always camel case. contentList, setContentList

**Review by tiagograf-AOT — APPROVED** [review]

**Review by djnunez-aot — APPROVED** [review]

File: met-web/src/components/engagement/form/AddSurveyBlockTabs.tsx (hunk 1 above)
Comment (VineetBala-AOT) [reply met-web/src/components/engagement/form/AddSurveyBlockTabs.tsx]: changed as per review comments

File: met-web/src/components/engagement/form/AddSurveyBlockTabs.tsx (hunk 3 above)
Comment (VineetBala-AOT) [reply met-web/src/components/engagement/form/AddSurveyBlockTabs.tsx]: changed

**Pull request merged**